### PR TITLE
Feat: add math extras

### PR DESCRIPTION
**Pull request by achamas-playco**
### Summary
🧮➕📝

This pull request adds the `@pixi/math-extras` package, which provides extra methods and functions for the `Point`, `ObservablePoint`, and `Rectangle` classes, as well as some utility functions for comparing and intersecting lines and segments. The pull request updates the `package.json` file to include the new subpath for importing the package, and modifies the existing classes and files to use the new features. The pull request also adds unit tests and documentation for the new package.

> _`math-extras` grows_
> _adding methods to `Point`, `Rect`_
> _autumn of vectors_

### Walkthrough
*  Add and initialize `@pixi/math-extras` package, which extends existing classes with additional features ([link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R138), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R233-R240), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-696d1b23b9b8ac4c207130e7614bd38daa4d92f03c23ae32c1068aa4bd09f09bL16-R16), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-bbeb34b2b08bbd70cefb5fab62c8cde66f251a0e7a33f1a3de45c129c3720579R7), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-e6eee386bceeec1834c42268cc214bb78d0e0359c7fe1bc8205ee611e6e6b0adR1-R7), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-913d0958bf230c12fddc3f1fa2176a0e03d15c6730f74cca5b86ec68e4f95e0cR1-R400), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-2b6c6ad16ce73a7e0543e8f6ee3ab922cf5382c00250434aabf8edab2ec55b6aR1-R124))
*  Augment type definitions of `Point`, `ObservablePoint`, and `Rectangle` classes with extra methods from `@pixi/math-extras` package ([link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-94822a5a1842d1f5d86d1c7c247f97ff642c36d13bc8716c312e602e5efcb741L1-R7), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-76ca6a86ff3ff7625b02d8d27a188464b3b4957781b05f0da2f6df73f39b30ecL2-R8), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-73ab825d7c964ace62a22b14f002262323217caac115dd30ba7b6c76d9d65bfaR11-R13))
*  Remove unused `mixin` methods and dependencies from `Point` and `ObservablePoint` classes ([link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-94822a5a1842d1f5d86d1c7c247f97ff642c36d13bc8716c312e602e5efcb741R34-R42), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-76ca6a86ff3ff7625b02d8d27a188464b3b4957781b05f0da2f6df73f39b30ecR18-R26))
*  Add unit tests for `floatEqual`, `lineIntersection`, and `segmentIntersection` functions from `@pixi/math-extras` package ([link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-5522348a1ce6256b47ac0ea48313bc29a5f12b01e35c3641fafd65c9f1067b4fR1-R14), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-85d6121488688a49b924867ed5c7569b227978952abdc587da9c9be8a043aa06R1-R72), [link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-59843ff8ac9ef165df6dee035ab6617bba200ed6de52565df5274700197e4fceR1-R85))
*  Add unit tests for extra methods added to `Rectangle` class by `@pixi/math-extras` package ([link](https://github.com/pixijs/pixijs/pull/9862/files?diff=unified&w=0#diff-115c13f73e10ba606b0631a269a487bd043cfc4533fe9855ebaf0a83ed0f9d18R1-R172))

### CODE REVIEW

**Review by GoodBoyDigital — APPROVED** [review]

File: src/environment/browser/browserAll.ts [hunk 1]
```diff
@@ -4,3 +4,4 @@ import '../../assets/init';
 import '../../events/init';
 import '../../spritesheet/init';
 import '../../rendering/init';
+import '../../math-extras/init';
```
Comment (Zyie) [inline src/environment/browser/browserAll.ts:7]: math-extraxs isn't included by default in v7, so this doesn't need to be here
```suggestion

```

File: src/math-extras/util.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+ * Computes the point where non-coincident and non-parallel Lines intersect.
+ * Coincident or parallel lines return a `NaN` point `{x: NaN, y: NaN}`.
+ * The intersection point may land outside the extents of the lines.
+ *
+ * _Note: Only available with **pixi.js/math-extras**._
+ * @param aStart - First point of the first line.
+ * @param aEnd - Second point of the first line.
+ * @param bStart - First point of the second line.
+ * @param bEnd - Second point of the second line.
+ * @returns {PointData} The point where the lines intersect.
```
Comment (Zyie) [inline src/math-extras/util.ts:118]:
```suggestion
 * @returns {PointData} The point where the lines intersect.
 * @memberof maths
```

File: src/math-extras/util.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+ * The intersection point may land outside the extents of the lines.
+ *
+ * _Note: Only available with **pixi.js/math-extras**._
+ * @param aStart - First point of the first line.
+ * @param aEnd - Second point of the first line.
+ * @param bStart - First point of the second line.
+ * @param bEnd - Second point of the second line.
+ * @param {PointData} outPoint - A Point-like object in which to store the value,
+ * optional (otherwise will create a new Point).
+ * @returns {PointData} The point where the lines intersect or a `NaN` Point.
```
Comment (Zyie) [inline src/math-extras/util.ts:133]:
```suggestion
 * @returns {PointData} The point where the lines intersect or a `NaN` Point.
 * @memberof maths
```

File: src/math-extras/util.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+ * Computes the point where non-coincident and non-parallel segments intersect.
+ * Coincident, parallel or non-intersecting segments return a `NaN` point `{x: NaN, y: NaN}`.
+ * The intersection point must land inside the extents of the segments or return a `NaN` Point.
+ *
+ * _Note: Only available with **pixi.js/math-extras**._
+ * @param aStart - Starting point of the first segment.
+ * @param aEnd - Ending point of the first segment.
+ * @param bStart - Starting point of the second segment.
+ * @param bEnd - Ending point of the second segment.
+ * @returns {PointData} The point where the segments intersect.
```
Comment (Zyie) [inline src/math-extras/util.ts:153]:
```suggestion
 * @returns {PointData} The point where the segments intersect.
 * @memberof maths
```

File: src/math-extras/util.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+ * The intersection point must land inside the extents of the segments or return a `NaN` Point.
+ *
+ * _Note: Only available with **pixi.js/math-extras**._
+ * @param aStart - Starting point of the first segment.
+ * @param aEnd - Ending point of the first segment.
+ * @param bStart - Starting point of the second segment.
+ * @param bEnd - Ending point of the second segment.
+ * @param {PointData} outPoint - A Point-like object in which to store the value,
+ * optional (otherwise will create a new Point).
+ * @returns {PointData} The point where the segments intersect or a `NaN` Point.
```
Comment (Zyie) [inline src/math-extras/util.ts:168]:
```suggestion
 * @returns {PointData} The point where the segments intersect or a `NaN` Point.
 * @memberof maths
```

File: src/maths/point/Point.ts [hunk 6]
```diff
@@ -1,7 +1,11 @@
 /* eslint-disable @typescript-eslint/no-use-before-define */
+// import type { Dict } from '../../utils/types';
```
Comment (Zyie) [inline src/maths/point/Point.ts:2]:
```suggestion
```

**Review by Zyie — CHANGES_REQUESTED** [review]

File: src/maths/point/ObservablePoint.ts [hunk 7]
```diff
@@ -1,6 +1,10 @@
+// import type { Dict } from '../../utils/types';
```
Comment (Zyie) [inline src/maths/point/ObservablePoint.ts:1]:
```suggestion
```

**Review by Zyie — APPROVED** [review]

**Pull request merged**